src/DI/ImageStorageExtension.php

```diff
@@ -2,66 +2,49 @@
 
 namespace Contributte\ImageStorage\DI;
 
+use Contributte;
+use Contributte\ImageStorage\ImageStorage;
+use Nette;
 use Nette\DI\CompilerExtension;
-use Nette\DI\Helpers;
+use Nette\Schema\Expect;
+use Nette\Schema\Schema;
 
 class ImageStorageExtension extends CompilerExtension
 {
 
-	/** @var mixed[] */
-	private $defaults = [
-		'data_path'          => '%wwwDir%/../public/data',
-		'data_dir'           => 'data',
-		'algorithm_file'     => 'sha1_file',
-		'algorithm_content'  => 'sha1',
-		'quality'            => 85,
-		'default_transform'  => 'fit',
-		'noimage_identifier' => 'noimage/03/no-image.png',
-		'friendly_url'       => false,
-	];
+	public function getConfigSchema(): Schema
+	{
+		return Expect::structure([
+			'data_path' => Expect::string('%wwwDir%/../public/data'),
```

Comment:
Schema does not automatically expand %parameter%, you need to do what I described ;)